integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/complexType/TestComplexDataType.scala

```diff
@@ -712,5 +713,166 @@ class TestComplexDataType extends QueryTest with BeforeAndAfterAll {
+          | salary Int
+          | )
+          | PARTITIONED BY (area array<string>)
+          | STORED BY 'carbondata'
+        """.stripMargin))
+    assertResult("Cannot use array<string> for partition column;")(arrayException.getMessage)
+    sql("DROP TABLE IF EXISTS list_table_area_origin")
+    val structException = intercept[AnalysisException](
+      sql("""
+            | CREATE TABLE IF NOT EXISTS list_table_area_origin
+            | (
+            | id Int,
+            | vin string,
+            | logdate Timestamp,
+            | phonenumber Long,
+            | country array<string>,
+            | salary Int
+            | )
+            | PARTITIONED BY (area struct<b:int>)
+            | STORED BY 'carbondata'
+          """.stripMargin)
+    )
+    assertResult("Cannot use struct<b:int> for partition column;")(structException.getMessage)
+  }
+
+  test("test block preaggregate") {
+    sql("DROP TABLE IF EXISTS test")
+    sql("create table test(id int,a struct<b:int>) stored by 'carbondata'")
+    sql("insert into test values (1,2)")
+    sql("insert into test values (1,2)")
+    sql("insert into test values (1,2)")
+    val structException = intercept[MalformedCarbonCommandException](
+      sql("create datamap preagg_sum on table test using 'preaggregate' as select id,sum(a.b) from test group by id"))
+    assertResult("Preaggregate is unsupported for ComplexData type column: a.b")(structException.getMessage)
+    sql("DROP TABLE IF EXISTS test")
+    sql("create table test(id int,a array<int>) stored by 'carbondata'")
+    sql("insert into test values (1,2)")
+    val arrayException = intercept[MalformedCarbonCommandException](
+      sql("create datamap preagg_sum on table test using 'preaggregate' as select id,sum(a[0]) from test group by id"))
+    assertResult("Preaggregate is unsupported for ComplexData type column: a[0]")(arrayException.getMessage)
+  }
+
+  test("test block dictionary exclude for child column") {
+    sql("DROP TABLE IF EXISTS table1")
+    sql(
+      "create table table1 (roll int,a struct<b:int,c:string,d:int,e:string,f:struct<g:int," +
+      "h:string,i:int>,j:int>) stored " +
+      "by " +
+      "'carbondata' tblproperties('dictionary_exclude'='a')")
+    sql("insert into table1 values(1,'1$abc$2$efg$3:mno:4$5')")
+    checkAnswer(sql("select a.b from table1"), Seq(Row(1)))
+    sql("DROP TABLE IF EXISTS table1")
+    val structException = intercept[MalformedCarbonCommandException](
+    sql(
+      "create table table1 (roll int,a struct<b:int,c:string,d:int,e:string,f:struct<g:int," +
+      "h:string,i:int>,j:int>) stored " +
+      "by " +
+      "'carbondata' tblproperties('dictionary_exclude'='a.b')"))
+    assertResult(
+      "DICTIONARY_EXCLUDE column: a.b does not exist in table or unsupported for complex child " +
+      "column. Please check create table statement.")(
+      structException.getMessage)
+    sql("DROP TABLE IF EXISTS table1")
```

Comment:
drop will not be executred if assertion fails. Please make sure all tables are deleted in afterAll also.